src/reconcile/rec_visibility.c

```diff
@@ -375,18 +371,6 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
             }
         }
         if (upd != NULL) {
-            /*
-             * If we're seeing a non-timestamped tombstone being applied on top of a timestamped
-             * update, force the tombstone to be globally visible so that we destroy the key. This
-             * isn't technically correct but when we mix timestamps, we're not guaranteeing that
-             * that older readers will be able to continue reading content that has been made
-             * invisible by a non-timestamped update.
-             */
-            if (upd_select->stop_ts == WT_TS_NONE && upd->start_ts != WT_TS_NONE)
```

Comment:
I don't think this is being preserved by your changes. If the tombstone is non-timestamped, we need to force the stop to be `(0, 0)` so the key gets blown way. If you simply force it to be the stop time pair, it's not guaranteed to be globally visible.

Reply:
What I thought would happen with a non-timestamped tombstone is that we'd set `upd_select->stop_ts = WT_TS_NONE` at line 358.  Then we wouldn't look for another update at line 368 because the tombstone is globally visible.  Then the new check down at line 428 would set `upd_select->start_ts = upd_select->stop_ts (= WT_TS_NONE)`.

Are you saying we need to do something special with `upd_select->stop_txn` as well?

Reply:
Actually I think it's fine. If that update later gets written to the history store, it'll trigger a deletion of the history store content for that key. I think the reason I was doing this was to avoid a malformed cell. But since you're repairing the time pairs, it's fine.

Reply:
Thanks, resolving.